source/tests/system/nidmm_driver_api_tests.cpp

```diff
@@ -0,0 +1,296 @@
+            dmm::CloseResponse response;
+
+            ::grpc::Status status = GetStub()->Close(&context, request, &response);
+
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+        }
+
+        std::string get_error_message(int error_status)
+        {
+            ::grpc::ClientContext context;
+            dmm::GetErrorMessageRequest request;
+            request.mutable_vi()->set_id(GetSessionId());
+            request.set_error_code(error_status);
+            dmm::GetErrorMessageResponse response;
+
+            ::grpc::Status status = GetStub()->GetErrorMessage(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            EXPECT_EQ(kDMMDriverApiSuccess, response.status());
+            return response.error_message();
+        }
+
+        ViBoolean get_bool_attribute(const char* channel_name, dmm::NiDMMAttributes attribute_id)
+        {
+            ::grpc::ClientContext context;
+            dmm::GetAttributeViBooleanRequest request;
+            request.mutable_vi()->set_id(GetSessionId());
+            request.set_channel_name(channel_name);
+            request.set_attribute_id(attribute_id);
+            dmm::GetAttributeViBooleanResponse response;
+            ::grpc::Status status = GetStub()->GetAttributeViBoolean(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+        ViInt32 get_int32_attribute(const char* channel_name, dmm::NiDMMAttributes attribute_id)
+        {
+            ::grpc::ClientContext context;
+            dmm::GetAttributeViInt32Request request;
+            request.mutable_vi()->set_id(GetSessionId());
+            request.set_channel_name(channel_name);
+            request.set_attribute_id(attribute_id);
+            dmm::GetAttributeViInt32Response response;
+            ::grpc::Status status = GetStub()->GetAttributeViInt32(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+        ViReal64 get_real64_attribute(const char* channel_name, dmm::NiDMMAttributes attribute_id)
+        {
+            ::grpc::ClientContext context;
+            dmm::GetAttributeViReal64Request request;
+            request.mutable_vi()->set_id(GetSessionId());
+            request.set_channel_name(channel_name);
+            request.set_attribute_id(attribute_id);
+            dmm::GetAttributeViReal64Response response;
+            ::grpc::Status status = GetStub()->GetAttributeViReal64(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+        std::string get_string_attribute(const char* channel_name, dmm::NiDMMAttributes attribute_id)
```

Comment:
Looks like this helper is not being used. Is there any plan to submit additional tests for string? If not, this helper can be removed.

Reply:
removed it